### PR TITLE
Fix failures due to invalid numpy scalars.

### PR DESCRIPTION
**Pull request by rossbar**
## What
Fixes the failures in `deepcell-tf` for numpy v1.24 by updating to use the 

## Why
Remove the upper bound on numpy.

Marking as draft for now, as this depends on vanvalenlab/deepcell-tracking#112 as well. The tracking tests in `deepcell-tf` will continue to fail until those changes make it into a release.

### CODE REVIEW

**Comment by rossbar** [comment]
Hmm I was expecting the tests to fail here due to the dependency on `deepcell-tracking`... not exactly sure why that's not the case.

**Comment by rossbar** [comment]
> Hmm I was expecting the tests to fail here due to the dependency on deepcell-tracking... not exactly sure why that's not the case.

To answer my own question: It's likely due to dependency pins in `deepcell-tracking`. Therefore, CI here would have to be run against `deepcell-tracking` after vanvalenlab/deepcell-tracking#112 is *released*.

**Review by msschwartz21 — COMMENTED** [review]
I reviewed the github action logs and it looks like we never actually test against `numpy` > 1.23.5. I'm not sure if that's an artifact of the pins in `deepcell_tracking`, something to do with caching the python environment or something else entirely.

Do you think we need to modify something in our tests to ensure we are catching 1.24 or just to check that everything works by running the tests locally with `numpy=1.24`?

**Comment by rossbar** [comment]
Yeah this was part of the motivation for adding the "test-against-dev" job to the `deepcell-tf` CI, but the transitive dependencies between all the deepcell- projects makes it difficult to track down exactly how the environment is being created.

I would like to try modifying the CI to get an updated version of all the dependencies, though it may end up being more effort than it's worth.

**Comment by rossbar** [comment]
Ok, as so often happens this turned out to be more complicated than it needed to be. In summary, I think this was caused by 2 main things:
 1. The trick I was relying on to `pip install` from the source repo in #636 wasn't actually working.
 2. I believe there is transitive (non-deepcell) dependency that is pinning numpy<1.24.

I took care of (1) by making the installation procedure in the `tests.yaml` workflow more explicit. (2) requires an extra step to force a numpy upgrade in the test-dev job. This will likely not be necessary for much longer, but it also shouldn't hurt anything either.

Long story short - CI is now *actually* running tests against dev versions of the other `deepcell-` libraries as well as the latest version of numpy.

**Review by msschwartz21 — APPROVED** [review]
